### ISSUE TITLE
PHP7 compatibility

### ISSUE THREAD

**Issue by propertunist** [hypeJunction/hypeCategories#41]
i am testing the latest version of this plugin with PHP7.0.7 and found that due to the changes with the way PHP processes function variable input types, i needed to remove the declaration of 'array' from the function 'setItemCategories' - since the value of this variable may either be an array or a string.
so the declaration on line 143 in categories.php is now:

```
public function setItemCategories(ElggEntity $entity, $categories = array(), array $options = array()) {
```

**Comment by propertunist** [comment]
additionally, the $return arrays passed into some of the routing hooks are being passed in as booleans but elgg_extract expects an array. so i have added this as a workaround:

```
    if (!is_array($return))
    {
        $return = array($return);
    }
```

**Comment by hypeJunction** [comment]
Re: categories. Type hinting is there for a reason, and passing a non-array may result in errors. So instead of hacking the declaration, you should probably update your uses of the function.
Re: route hooks. Routing hooks should bail if return value is not array, as it most likely indicates that another hook has already output a page.

**Comment by propertunist** [comment]
yes, i appreciate more needs to change than my workaround change - i just needed to bypass the problem quickly as i had upgraded my production site to php7. i am not using the function in any of my code, the situation arose just through standard use of hypeCategories on it's own (as far as i am aware). 
the route hooks that need updates are:
function routeCategoryContextPages
function storeFinalRoute

so are you saying that i need to replace:

```
        if (!is_array($return))
        {
            $return = array($return);
        }
```

with 

```
        if (!is_array($return))
        {
            return;
        }
```

?
